api/envoy/config/cluster/v3/cluster.proto

```diff
@@ -319,6 +319,31 @@ message Cluster {
     // The number of random healthy hosts from which the host with the fewest active requests will
     // be chosen. Defaults to 2 so that we perform two-choice selection if the field is not set.
     google.protobuf.UInt32Value choice_count = 1 [(validate.rules).uint32 = {gte: 2}];
+
+    // The following formula is used to calculate the dynamic weights when hosts have different load
+    // balancing weights:
+    //
+    // `weight = load_balancing_weight / (active_requests + 1)^active_request_bias`
+    //
+    // The larger the active request bias is, the more aggressively active requests will lower the
+    // effective weight when all host weights are not equal.
+    //
+    // `active_request_bias` must be greater than or equal to 0.0.
+    //
+    // When `active_request_bias == 0.0` the Least Request Load Balancer doesn't consider the number
+    // of active requests at the time it picks a host and behaves like the Round Robin Load
+    // Balancer.
+    //
+    // When `active_request_bias > 0.0` the Least Request Load Balancer scales the load balancing
+    // weight by the number of active requests at the time it does a pick.
+    //
+    // The value is cached for performance reasons and refreshed whenever one of the Load Balancer's
+    // host sets changes, e.g., whenever there is a host membership update or a host load balancing
+    // weight change.
+    //
+    // .. note::
+    //   This setting only takes effect if all host weights are not equal.
+    core.v3.RuntimeDouble active_request_bias = 2;
```

Comment:
So this must be the cause of the bloat.

I'm somewhat familiar with the Runtime infrastructure but not deeply so. If we have 10k clusters would you runtime-override the active_request_bias for a subset of them? Or is the runtime override supposed to be a kill-switch to broadly disable a feature?

Maybe we are paying a lot for functionality that we don't need? WDYT?

BTW I'm fine if we want to submit this temporarily and then reconsider how we can better represent the flexibility we want from the Runtime system. @alyssawilk would be curious for your take here. Thank you!

Reply:
Captured this question in https://github.com/envoyproxy/envoy/issues/11872

Reply:
not sure if this addresses the problem but I'd encourage using xds for changing per-cluster values rather than using runtime overrides.  IMO runtime changes make some sense for global things which you want fast changes on (DoS protection knobs) but something like this could be done by CDS.

Reply:
I just pushed a commit that uses `double` instead of `RuntimeDouble` to see whether not using `RuntimeDouble` reduces the memory overhead — I'm waiting for the CI results to see how things go on Linux, but it did not help on macOS.

As far as I can tell the helper used by the memory usage tests to create clusters doesn't create any instances of `LeastRequestLbConfig` message:

https://github.com/envoyproxy/envoy/blob/e71e4dc4204450175644fa89639f8046872f548f/test/integration/stats_integration_test.cc#L173-L203

So I don't understand why the per-cluster memory footprint increases.

Reply:
The [new CI result](https://github.com/envoyproxy/envoy/pull/11252/checks?check_run_id=843013408) shows that the per-cluster memory footprint of the integration tests using `double` is the same as using `RuntimeDouble`.

`RuntimeDouble` does contain  a string (`runtime_key`), so I do expect it to use a little more memory for clusters that specify a bias.

Making it runtime overridable would make it easier to change the bias without having to update the xDS server. But any of the approaches would be enough to improve the load balancing behavior during squeeze (fka red line) tests at Lyft, so I don't have a very strong opinion towards any of them.

@tonya11en and @mattklein123 said in [this Slack thread](https://envoyproxy.slack.com/archives/C78HA81DH/p1591996250269300) that they preferred to use `RuntimeDouble`, but we were only considering the potential performance overhead and not the memory overhead.

@jmarantz / @alyssawilk: given the new data do you still believe that it would be better not to use `RuntimeDouble`?